PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/cosmosserverlessspark/spark/ui/livy/batch/KillCosmosServerlessSparkBatchJobAction.kt

```diff
@@ -41,44 +41,48 @@ import java.io.IOException
+                            } else {
+                                return@flatMap account.killSparkBatchJobRequest(job.id().toString())
+                                        .doOnNext { resp ->
+                                            if (resp.code >= 300) {
+                                                val errMsg = "Failed to kill spark job ${respBatchJob.name()}. ${resp.message}"
+                                                PluginUtil.displayErrorDialog("Kill Serverless Spark Job", errMsg)
+                                                EventUtil.logErrorWithComplete(operation, ErrorType.serviceError, IOException(errMsg), null, null)
+                                            } else {
+                                                PluginUtil.displayInfoDialog("Kill Serverless Spark Job", "Successfully killed Spark Job ${respBatchJob.name()}!")
+                                                EventUtil.logEventWithComplete(EventType.info, operation, null, null)
+                                            }
+                                        }
+                                        .map { respBatchJob }
+                            }
+                        }
+                        .subscribe(
+                                {},
+                                { err ->
+                                    val errMsg = "Failed to kill spark job ${job.name()}. ${err.message}"
                                     PluginUtil.displayErrorDialog("Kill Serverless Spark Job", errMsg)
+                                    log().warn("Failed to kill serverless spark job. " + ExceptionUtils.getStackTrace(err))
                                     EventUtil.logErrorWithComplete(operation, ErrorType.serviceError, IOException(errMsg), null, null)
-                                } else {
-                                    PluginUtil.displayInfoDialog("Kill Serverless Spark Job", "Successfully killed Spark Job ${respBatchJob.name()}!")
-                                    EventUtil.logEventWithComplete(EventType.info, operation, null, null)
                                 }
-                            }
-                            .map { respBatchJob }
-                    }
-                }
-                .subscribe(
-                    {},
-                    { err ->
-                        val errMsg = "Failed to kill spark job ${job.name()}. ${err.message}"
-                        PluginUtil.displayErrorDialog("Kill Serverless Spark Job", errMsg)
-                        log().warn("Failed to kill serverless spark job. " + ExceptionUtils.getStackTrace(err))
-                        EventUtil.logErrorWithComplete(operation, ErrorType.serviceError, IOException(errMsg), null, null)
-                    }
-                )
+                        )
+            }
+            return false
+        } catch (ignored: RuntimeException) {
+            return true
```

Comment:
do we need to log the exception locally in logs?

Reply:
We have logged the exceptions in code. So, needn't here.